app/util/grids.ts

```diff
@@ -0,0 +1,64 @@
+import { ServerResponse } from 'http';
+import { NextFunction } from 'express';
+import HarmonyRequest from '../models/harmony-request';
+import { getGridsByName } from './cmr';
+import parseCRS from './crs';
+import { RequestValidationError } from './errors';
+import { keysToLowerCase } from './object';
+
+/**
+ * Throws an error if the grid name parameter is included along with regridding parameters.
+ *
+ * @param query - the query for the request
+ * @throws RequestValidationError if the combination of parameters is invalid.
+*/
+export function validateNoConflictingGridParameters(query: Record<string, unknown>): void {
+  const keys = keysToLowerCase(query);
+  if (keys.grid) {
+    if (keys.outputcrs || keys.scaleextent || keys.scalesize || keys.width || keys.height) {
+      throw new RequestValidationError('When including a grid query parameter, the following parameters may not be provided: scaleExtent, scaleSize, outputCrs, height, and width.');
+    }
+  }
+}
+
+/**
+ * Middleware to handle the grid parameter in a Harmony query, adding it to the DataOperation
+ * if necessary.
+ *
+ * @param operation - the DataOperation for the request
```

Comment:
nit: The JSDoc comment param list needs to be updated to reflect the real parameter list.